### PR TITLE
add an administration client 

### PR DESCRIPTION
**Pull request by jbtrystram**
built on top of #6

### CODE REVIEW

File: src/tokens/v1/client.rs [hunk 1]
```diff
@@ -0,0 +1,92 @@
+use super::data::*;
+use crate::error::ClientError;
+use crate::openid::TokenProvider;
+use crate::util::Client;
+use std::fmt::Debug;
+use tracing::instrument;
+use url::Url;
+
+/// A device registry client, backed by reqwest.
```
Comment (lulf) [inline src/tokens/v1/client.rs:9]: Outdated comment

File: src/util/client/mod.rs [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+/// A drogue HTTP client, backed by reqwest.
+
+#[async_trait]
+pub trait Client<TP>
+where
+    TP: TokenProvider,
+{
+    /// Constructor
+    fn new(client: reqwest::Client, url: Url, token_provider: TP) -> Self;
```
Comment (lulf) [inline src/util/client/mod.rs:20]: I find it a bit strange to have the constructor as part of the trait, as you normally know the type at construction time, and there is no need for restricting the constructor signature I think.

**Review by lulf — COMMENTED** [review]
I think it's a great set of refactorings @jbtrystram. Re; the roles I'm not sure if we discussed it earlier, but it looks good to me. Left a few comments in code.

File: src/util/client/mod.rs (hunk 2 above)
Comment (jbtrystram) [reply src/util/client/mod.rs:20]: yep I agree. It also force the user to import the trait as well, which is annoying. I removed it on the other PR :)

**Pull request merged**